src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -82,6 +87,23 @@ static CurlHandler()
             // Feature detection
             s_supportsSSL = (CurlFeatures.CURL_VERSION_SSL & s_curlVersionInfoData.features) != 0;
             s_supportsAutomaticDecompression = (CurlFeatures.CURL_VERSION_LIBZ & s_curlVersionInfoData.features) != 0;
+
+            // We can send any pointer, even null seems to work. But sending a non-null value, just in case
+            if (CurlSslBackend.AllowsSslCallback(s_curlVersionInfoData.ssl_version))
+            {
+                s_clientCertificatePolicy = ClientCertificatePolicy.Callback;
+                VerboseTrace("openSSL compatible");
+            }
+            else if (CurlSslBackend.AllowsClientCertCommonName(s_curlVersionInfoData.ssl_version))
```

Comment:
This support sort of confuses me.  How is the user supposed to facilitate it?  If they've loaded a cert from a PFX and don't have it in the OSX store, what's the story?  Worse, what if they loaded a cert from a PFX, and have a certificate with the same CN value but from a different issuer?  What happens then?

If we don't have a good story here we shouldn't try this integration path.